syntax = "proto3";

package tensorflow;
option cc_enable_arenas = true;
option java_outer_classname = "ConfigProtos";
option java_multiple_files = true;
option java_package = "org.tensorflow.framework";
// add go_package externally with copybara
//import "tensorflow/core/framework/cost_graph.proto";
//import "tensorflow/core/framework/graph.proto";
import "tensorflow/core/framework/step_stats.proto";
//import "tensorflow/core/protobuf/debug.proto";
//import "tensorflow/core/protobuf/cluster.proto";
//import "tensorflow/core/protobuf/rewriter_config.proto";

// Metadata output (i.e., non-Tensor) for a single Run() call.
message RunMetadata {
  // Statistics traced for this step. Populated if tracing is turned on via the
  // "RunOptions" proto.
  // EXPERIMENTAL: The format and set of events may change in future versions.
  StepStats step_stats = 1;

  // The cost graph for the computation defined by the run call.
//  CostGraphDef cost_graph = 2;

  // Graphs of the partitions executed by executors.
//  repeated GraphDef partition_graphs = 3;
}

message TraceDataProto {
    map<int64, Traces> traced_steps = 1;
}

message Traces {
    repeated RunMetadata traces = 1;
}

